src/Oro/Bundle/UserBundle/Controller/UserController.php

```diff
@@ -87,9 +88,9 @@ public function createAction()
      * @Template
      * @AclAncestor("pim_user_user_edit")
      */
-    public function updateAction(User $entity)
+    public function updateAction($id)
```

Comment:
We could create a ParamConverter who deal with extended entities and typehint with the interface or an option in the ParamConverter annotation. I'll make a proof of concept soon about this in order to remove this kind of stuff from controllers